samples/bluetooth/peripheral_hr/src/main.c

```diff
@@ -122,10 +117,50 @@ static void hrs_notify(void)
 	}
 }
 
+/* The devicetree node identifier for the "led0" alias. */
+#define LED0_NODE DT_ALIAS(led0)
+
+#if DT_NODE_HAS_STATUS(LED0_NODE, okay)
+#include <zephyr/drivers/gpio.h>
+#define HAS_LED     1
+static const struct gpio_dt_spec led = GPIO_DT_SPEC_GET(LED0_NODE, gpios);
+#define BLINK_ONOFF K_MSEC(500)
+
+static struct k_work_delayable blink_work;
+static bool                  led_is_on;
```

Comment:
`led_is_on` seems to be modify by multiple threads. Should be `volatile` or have some sort of thread safety added

Reply:
The work is stopped (or not started) before the variable is modifed and hence it is safe.